lib/kb_tophat2/Utils/TopHatUtil.py

```diff
@@ -586,20 +586,16 @@ def run_tophat2_app(self, params):
                                                        params.get('workspace_name'))
         genome_index_files = os.listdir(genome_index_file_dir)
 
-        log('generated genome index files: {}'.format(genome_index_files)) 
-        genome_index_file = filter(re.compile(".*\.\d\..*").match, genome_index_files)[0]
-        if re.match('.*\.rev\.\d\..*', genome_index_file):
-            genome_index_file_prefix = genome_index_file.split('rev')[0][:-1]
-        else:
-            genome_index_file_prefix = ''
-            for prefix in genome_index_file.split('.'):
-                if prefix.isdigit():
-                    break
-                else:
-                    genome_index_file_prefix += '.' + prefix
-            genome_index_file_prefix = genome_index_file_prefix[1:]
-
-        genome_index_base = genome_index_file_dir + '/' + genome_index_file_prefix
+        log('generated genome index files: {}'.format(genome_index_files))
+        genome_index_base = ""
+        for index_file in genome_index_files:
+            match = re.match('(.*)\.(rev\.)?\d\.\w+', index_file)
```

Comment:
One thing that always helps for regexes is to add a comment or two with example strings of what this regex will successfully match